tests/test_token_gen.py

```diff
@@ -614,6 +665,8 @@ def test_certificate_request_failure(self, user_mgt_app):
 class TestCertificateCaching:
 
     def test_certificate_caching(self, user_mgt_app, httpserver):
+        if _is_emulated():
```

Comment:
Yes, right now I don't have the understanding of the emulator required to know which of these tests are actually important for the emulator - what behaviour is supported and expected and what behaviour is undefined. The failing tests are all marked skipped for the emulator case here for review (so that the reviewer doesn't have to jump back and forth between test results and code). If somebody can triage them and note the important ones, I can then prioritise fixing those.

Reply:
Ideally everything except a few should pass on both modes. Some of the ones that are not relevant in the emulator mode are:

* test_sign_with_iam
* test_sign_with_iam_error
* test_sign_with_discovered_service_account
* test_sign_with_discovery_failure
* test_certificate_request_failure
* test_certificate_caching